File: api/models/protocol.go
package models

type Protocol string

const (
	// ProtocolHttpJson protocol to be used when deploying model as HTTP/JSON server
	ProtocolHttpJson Protocol = "HTTP_JSON"
	// ProtocolUpiV1 protocol to be used when deploying UPI-compatible model
	ProtocolUpiV1    Protocol = "UPI_V1"
)

File: python/sdk/merlin/protocol.py
import enum


class Protocol(enum.Enum):
    """
    Model deployment protocol.

    HTTP_JSON = Deploy model and expose HTTP server
    UPI_V1 = Deploy model and expose gRPC server compatible with universal-prediction-interface (https://github.com/caraml-dev/universal-prediction-interface)
    """
    HTTP_JSON = "HTTP_JSON"
    UPI_V1 = "UPI_V1"
